### PR TITLE
[UI] add selectable tags dropdown

### PR DESCRIPTION
**Pull request by BurningTreeC**
this now uses the new widget subclassing mechanism and the new tm-focus-selector message in combination with the new escapecss filter for focusing

if this gets accepted, I'll add PR's for the sidebar search and other dropdowns if desired

### CODE REVIEW

**Comment by Jermolene** [comment]
Hi @BurningTreeC thank you this looks good. There's a lot to think about, though, so I'd prefer to leave it until after v5.1.20.

**Comment by BurningTreeC** [comment]
>There's a lot to think about, though, so I'd prefer to leave it until after v5.1.20.

That's right, no problem :+1:

**Comment by BurningTreeC** [comment]
To make this happen for 5.1.21 ... what are the design questions/issues? I'd really like to make this as simple as possible

**Comment by BurningTreeC** [comment]
closing this PR because it has too many conflicts and I'll create a new one after the 5.1.20 release

**Comment by BurningTreeC** [comment]
This PR is ready for testing

**Comment by diego898** [comment]
🎉🎊

**Comment by BurningTreeC** [comment]
Hi @Jermolene, what do you think about this PR? There are some conflicts that are to be resolved and I'd probably create a new, cleaner, refactored PR but before I'll do so I'd like to hear your opinion if it's worth the work

File: core/modules/widgets/subclasses/edit-text-subclass.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+exports.prototype.execute = function(event) {
+	// Call the base class execute function
+	Object.getPrototypeOf(Object.getPrototypeOf(this)).execute.call(this,event);
+	this.refreshTiddler = this.getAttribute("refreshTiddler");
+	this.refreshAction = this.getAttribute("refreshAction");
+	this.saveTiddler = this.getAttribute("saveTiddler");
+};
+
+exports.prototype.refresh = function(event) {
+	var changedAttributes = this.computeAttributes();
```
Comment (pmario) [inline core/modules/widgets/subclasses/edit-text-subclass.js]: IMO if you'd use 

```
var cA = this.computeAttributes();
```

The `else if (...` would be much easier to read.

File: core/language/en-GB/Buttons.multids [hunk 2]
```diff
@@ -53,6 +53,11 @@ Import/Caption: import
 Import/Hint: Import many types of file including text, image, TiddlyWiki or JSON
 Info/Caption: info
 Info/Hint: Show information for this tiddler
+Input/Accept/Hint: Accept the selected ~ListItem
+Input/Clear/Hint: Clear the input field
+Input/Down/Hint: Select the next ~ListItem
+Input/Up/Hint: Select the previous ~ListItem
```
[the remark this answers is not in the document]
Comment (BurningTreeC) [reply core/language/en-GB/Buttons.multids]: @pmario that's true, I'll change it

File: core/modules/widgets/subclasses/edit-text-subclass.js (hunk 1 above)
Comment (BurningTreeC) [reply core/modules/widgets/subclasses/edit-text-subclass.js]: imo this is compliant with the core

File: core/wiki/macros/selectable-inputs.tid [hunk 3]
```diff
@@ -0,0 +1,95 @@
+title: $:/core/macros/selectable-inputs
+tags: $:/tags/Macro
+
+\define change-input-tab(stateTitle,tag,beforeafter,defaultState)
```
[the remark this answers is not in the document]
Comment (BurningTreeC) [reply core/wiki/macros/selectable-inputs.tid]: I'll add a whitespace trim where parts of the UI are created. For other macros it's unnecessary or what do you think?

[the remark this answers is not in the document]
Comment (pmario) [reply core/wiki/macros/selectable-inputs.tid]: It should be added, if needed. If macros are 2 or 4 lines, I think it's overkill.

**Pull request closed without merging**